packages/std/src/api.rs

```diff
@@ -130,163 +114,71 @@ impl From<StdError> for ApiError {
+            SystemError::NoSuchContract { addr } => write!(f, "No such contract: {}", addr),
+            SystemError::Unknown {} => write!(f, "Unknown system error"),
         }
     }
 }
 
-impl From<SystemError> for ApiSystemError {
-    fn from(value: SystemError) -> Self {
-        match value {
-            SystemError::InvalidRequest { error, .. } => ApiSystemError::InvalidRequest { error },
-            SystemError::NoSuchContract { addr, .. } => ApiSystemError::NoSuchContract { addr },
-            SystemError::Unknown { .. } => ApiSystemError::Unknown {},
-        }
-    }
-}
+pub type SystemResult<T> = Result<T, SystemError>;
 
 #[cfg(test)]
-mod test_result {
+mod test {
+    use snafu::ResultExt;
+
     use super::*;
-    use crate::errors::{contract_err, NoSuchContract, StdResult};
+    use crate::errors::{
+        contract_err, dyn_contract_err, invalid, unauthorized, Base64Err, NotFound, NullPointer,
+        SerializeErr, StdResult,
+    };
     use crate::serde::{from_slice, to_vec};
 
-    #[test]
-    fn convert_ok_result() {
-        let input: StdResult<Vec<u8>> = Ok(b"foo".to_vec());
-        let convert: ApiResult<Vec<u8>> = input.into();
-        assert_eq!(convert, ApiResult::Ok(b"foo".to_vec()));
+    fn assert_conversion(r: StdResult<()>) {
+        let error = r.unwrap_err();
+        let msg = format!("{}", error);
+        let converted: ApiError = error.into();
+        assert_eq!(msg, format!("{}", converted));
+        let round_trip: ApiError = from_slice(&to_vec(&converted).unwrap()).unwrap();
+        assert_eq!(round_trip, converted);
     }
 
     #[test]
-    fn check_ok_into_conversion() {
-        let input: StdResult<bool> = Ok(true);
-        let convert: ApiResult<i32> = input.into();
-        assert_eq!(convert, ApiResult::Ok(1i32));
-        let expanded: Result<i64, ApiError> = convert.into();
-        assert!(expanded.is_ok());
-        assert_eq!(expanded.unwrap(), 1i64);
+    fn to_api_result_works_for_ok() {
+        let input: StdResult<Vec<u8>> = Ok(b"foo".to_vec());
+        assert_eq!(to_api_result(input), ApiResult::Ok(b"foo".to_vec()));
     }
 
     #[test]
-    fn convert_err_result() {
+    fn to_api_result_works_for_err() {
         let input: StdResult<()> = contract_err("sample error");
-        let convert: ApiResult<()> = input.into();
         assert_eq!(
-            convert,
+            to_api_result(input),
             ApiResult::Err(ApiError::ContractErr {
                 msg: "sample error".to_string()
             })
         );
-        let reconvert: Result<(), ApiError> = convert.into();
-        match reconvert {
-            Ok(_) => panic!("must be error"),
-            Err(e) => assert_eq!(
-                e,
-                ApiError::ContractErr {
-                    msg: "sample error".to_string()
-                }
-            ),
-        }
-    }
-
-    #[test]
```

Comment:
All these deleted test cases are not needed?

Reply:
Jap. What was deletes are the now unnecessary conversions
- `ApiResult` -> `Result<T, ApiError>`
- SystemError/ApiSystemError conversions